docs/standards.md

```diff
@@ -197,15 +198,140 @@ Things to take note of:
+
+On the other end of the account-hacker link, [hacker.util.js](../tests/util/hacker.test.util.js) contains the hacker data in the form of hacker objects. These hacker objects have an `accountId` attribute which references an account's `_id`. The matching between the hacker object and the respective account object it needs to link to is also done by nomenclature. For example, a hacker on a team would be called `TeamHackerX` where X is a number. This hacker's account object would be within the array specified by `hackerAccounts.stored.team`. The specific account object is referenced by its index in the array. That index is the same as the value X in the name of the hacker object.
 
 ### Validation files
 
-`TODO`
+Validation files are located in the `middlewares` directory under a subdirectory called `validators`. The filenames are `<X>.validator.js`. To be used in a route, a validator needs to be a list express validator functions. We have generic validator functions located in [validator.helper.js](../middlewares/validators/validator.helper.js). The goal is to have a generic validator function for each datatype, which can then be put in the list of validators as the situation requires. The generic validators may take several arguments, but generally require:
+  * Where the variable is located: `body`, `query`, `header`, or `param`. To validate a value located in `req.body`, one should use `body`
+  * The fieldname, such as `email` for `req.body.email`.
+  * A boolean detailing whether this value is optional. `false` means that the value is necessary.
+
+A validator example:
+```javascript
+"use strict";
+const VALIDATOR = require("./validator.helper");
+const Constants = require("../../constants/general.constant");
+
+module.exports = {
+    inviteAccountValidator: [
+        VALIDATOR.regexValidator("body", "email", false, Constants.EMAIL_REGEX),
+        VALIDATOR.enumValidator("body", "accountType", Constants.EXTENDED_USER_TYPES, false)
+    ]
+};
+```
+
+A route would use a validator in the following manner: 
+```javascript
+    accountRouter.route("/:id").patch(
+        ...
+        // validators
+        Middleware.Validator.RouteParam.idValidator,
+        Middleware.Validator.Account.updateAccountValidator,
+
+        Middleware.parseBody.middleware,
+        ...
+    );
+```
+
+`Middleware.parseBody.middleware` evaluates the validator functions, so it is generally placed right after the validators. If there is an error during validation, it will send a 422 error status to be handled. If all the validators pass, then it will move the validated values into req.body.
```

Comment:
> it will move the validated values into req.body.

Note that it will overwrite anything that is in `req.body`. So, if `req.body` is the following:
```json
{
  "A": "foo",
  "B": true,
  "C": { 
    "bar": "baz" 
  }
}
```
and we validate for only `A`, and `B`, then the `req.body` will have the following form after validation:
```json
{
  "A": "foo",
  "B": true
}
```